ncm-spma/src/main/pan/components/spma/schema.pan

```diff
@@ -4,18 +4,14 @@
 
 declaration template components/spma/schema;
 
+include 'pan/legacy';
 include 'quattor/types/component';
 include 'components/spma/functions';
-include 'components/spma/ips/schema';
-include 'components/spma/yum/schema';
 include 'components/spma/software';
 
-type component_spma_type = {
-    include structure_component
-    include component_spma_ips
-    include component_spma_yum
+type component_spma_common = {
     "cmdfile"       : string = "/var/tmp/spma-commands" # where to save commands for spma-run script
-    "packager"      : string = "yum" with match (SELF, '^(yum|ips|rpm)$') # system packager to be used (yum,ips,rpm)
+    "packager"      : string = "yum" with match (SELF, '^(yum|yumng|ips)$') # system packager to be used
```

Comment:
The way you switch between yum and yumng is you set the packager to yum and then set the "suffix" attribute to ng. The argument is that this keeps "packager" correct (it's still yum) but allows multiple variants of the packager (thinking ahead to DNF). Hopefully @jnovy can explain his idea in more detail.